astroquery/solarsystem/pds/core.py

```diff
@@ -0,0 +1,406 @@
+        request_payload = dict(
+            [
+                ("abbrev", planet[:3]),
+                ("ephem", planet_defaults[planet]["ephem"]),
+                ("time", epoch.utc.iso[:16]),
+                ("fov", 10),  # next few are figure options, can be hardcoded and ignored
+                ("fov_unit", planet.capitalize() + " radii"),
+                ("center", "body"),
+                ("center_body", planet.capitalize()),
+                ("center_ansa", planet_defaults[planet]["center_ansa"]),
+                ("center_ew", "east"),
+                ("center_ra", ""),
+                ("center_ra_type", "hours"),
+                ("center_dec", ""),
+                ("center_star", ""),
+                ("viewpoint", viewpoint),
+                ("observatory", "Earth's center"),  # has no effect if viewpoint != observatory so can hardcode. no plans to implement calling observatories by name since ring node only names like 8 observatories
+                ("latitude", latitude),
+                ("longitude", longitude),
+                ("lon_dir", "east"),
+                ("altitude", altitude),
+                ("moons", planet_defaults[planet]["moons"]),
+                ("rings", planet_defaults[planet]["rings"]),
+                ("arcmodel", neptune_arcmodels[int(neptune_arcmodel)]),
+                ("extra_ra", ""),  # figure options below this line, can all be hardcoded and ignored
+                ("extra_ra_type", "hours"),
+                ("extra_dec", ""),
+                ("extra_name", ""),
+                ("title", ""),
+                ("labels", "Small (6 points)"),
+                ("moonpts", "0"),
+                ("blank", "No"),
+                ("opacity", "Transparent"),
+                ("peris", "None"),
+                ("peripts", "4"),
+                ("arcpts", "4"),
+                ("meridians", "Yes"),
+                ("output", "html"),
+            ]
+        )
+
+        # return request_payload if desired
+        if get_query_payload:
+            return request_payload
+
+        # query and parse
+        response = self._request(
+            "GET", self._url, params=request_payload, timeout=self._timeout, cache=cache
+        )
+
+        return response
+
+    def _parse_result(self, response, verbose=None):
+        """
+        Routine for parsing data from ring node
+
+        Parameters
+        ----------
+        self : RingNodeClass instance
+        response : list
+            raw response from server
+
+
+        Returns
+        -------
+        bodytable : `astropy.QTable`
+        ringtable : `astropy.QTable`
+        """
+        self.last_response = response
+        try:
+            self._last_query.remove_cache_file(self.cache_location)
+        except FileNotFoundError:
+            # this is allowed: if `cache` was set to False, this
+            # won't be needed
+            pass
```

Comment:
You have taken this snippet of code from the `jplhorizons` sub-package, but you haven't copied the following test code:
 https://github.com/astropy/astroquery/blob/be0938f9bd9a989e25cf98d255635278ad883633/astroquery/jplhorizons/tests/test_jplhorizons.py#L62-L66 This must be the cause of the unexpected `AttributeError` failures in the tests.

Reply:
ok, I added this test.  it doesn't seem to solve the problem, though.  I'm not sure how to diagnose this especially because all the tests pass for me. any ideas?  Thanks again for all your help with this, by the way

Reply:
how would I modify `patch_request` to set the `_last_query`? 

that does seem like the way to go because otherwise I would no longer be able to use the `pds.RingNode.ephemeris()` syntax and instead would need to instantiate a `RingNode` object, then set `_last_query`, and finally call `ephemeris()`.  Yes, I am running tests with tox, which is why it's so confusing that it works on my machine... maybe I need to use different tox options.  BTW, I have been trying to model all the monkeypatch stuff off of what the jplhorizons astroquery tool did, and I'm not sure what I am missing.

Reply:
I see these failures when running only the local tests. When the remotes are also run all tests pass, showing that something is off with the mocking. I'm investigating of how it's all fine for jplhorizons but going off here.